src/libImaging/Mode.h

```diff
@@ -0,0 +1,240 @@
+#ifndef __MODE_H__
+#define __MODE_H__
+
+typedef enum {
+    IMAGING_MODE_UNKNOWN,
+
+    IMAGING_MODE_1,
+    IMAGING_MODE_CMYK,
+    IMAGING_MODE_F,
+    IMAGING_MODE_HSV,
+    IMAGING_MODE_I,
+    IMAGING_MODE_L,
+    IMAGING_MODE_LA,
+    IMAGING_MODE_LAB,
+    IMAGING_MODE_La,
+    IMAGING_MODE_P,
+    IMAGING_MODE_PA,
+    IMAGING_MODE_RGB,
+    IMAGING_MODE_RGBA,
+    IMAGING_MODE_RGBX,
+    IMAGING_MODE_RGBa,
+    IMAGING_MODE_YCbCr,
+
+    IMAGING_MODE_BGR_15,
+    IMAGING_MODE_BGR_16,
+    IMAGING_MODE_BGR_24,
+
+    IMAGING_MODE_I_16,
+    IMAGING_MODE_I_16L,
+    IMAGING_MODE_I_16B,
+    IMAGING_MODE_I_16N,
+    IMAGING_MODE_I_32L,
+    IMAGING_MODE_I_32B,
+} ModeID;
+
+typedef struct {
+    const char *const name;
+} ModeData;
```

Comment:
You've created a struct with only one member, presumably because you expect to add more members in the future. Are you trying to prepare for #6547?

Reply:
There are some things I think could be added to this struct in the future, but the more important reason for this as it is right now is type safety. A `char *` could be almost anything, but a `ModeData` can only be a `ModeData`.

Reply:
Correct me if I'm wrong, but the only place where `ModeData` is returned is from Mode.c's `getModeData()` (and similarly, `RawModeData` from `getRawModeData()`)? It's used there, and in the construction of the `MODES` array in Mode.c, and by design, never anywhere else? I don't think we even pass between `ModeData` between functions - every use outside of Mode.c is just to access the `name`.

So the only benefit is to assure us that `getModeData()` and `getRawModeData()` return the correct output? This could just be my opinion, but I think they are simple functions, and it's overkill to be passing around a struct to the rest of the C code to ensure their correctness.